jwst/lib/basic_utils.py

```diff
@@ -1,9 +1,61 @@
 """General utility objects"""
 
+from stdatamodels.jwst import datamodels
 from stdatamodels.jwst.datamodels import dqflags
 import numpy as np
 
 
+def use_datamodel(input, model_class=None):
```

Comment:
This method is used in place of `datamodels.open` which is often used in a `with` statement. When this method doesn't clone, convert or open the input file this will lead to the input file being closed at the end of the `with` statement.

Reply:
this is not an issue since the returned object is either a new instance of the input model or a modified copy.

Reply:
Here's a minimal reproducer for the issue:
```python
import jwst.datamodels as dm
from jwst.reset import ResetStep

m = dm.RampModel()
m.meta.instrument.detector = 'NRCA1'
m.save('foo.fits')

m = dm.open('foo.fits')
step = ResetStep()
step(m)
m.info()
```
Running this with jwst main shows the expected output from `m.info`. Running the example with this branch results in:
```
OSError: Cannot access data from closed ASDF file
```
Because the step closes the input file (even though the step is not part of a pipeline).
```

Reply:
thanks for this! I fixed this and now it behaves as expected